.github/workflows/main.yml

```diff
@@ -13,19 +13,19 @@ jobs:
   test:
     strategy:
       matrix:
-        os: ['ubuntu-18.04', 'ubuntu-20.04', 'ubuntu-22.04']
+        os: ['ubuntu-20.04', 'ubuntu-22.04']
 
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v2 # respository path: $GITHUB_WORKSPACE
-      - uses: actions/setup-go@v2
+      - uses: actions/checkout@v4 # respository path: $GITHUB_WORKSPACE
+      - uses: actions/setup-go@v5
         with:
-          go-version: 1.16.2
+          go-version: 1.21.7
 
       - name: Install golint
-        run: go get -u golang.org/x/lint/golint
+        run: go install golang.org/x/lint/golint@latest
```

Comment:
You could use golangci for this - it'd be a superset of golint, and there's a public action for it.

https://github.com/golangci/golangci-lint